src/components/UserProgress.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import LinearProgress from '@material-ui/core/LinearProgress'
+import { makeWidthFlexible, XYPlot, XAxis, YAxis, LineSeries, MarkSeries } from 'react-vis'
+import { COLOR } from '../constants/Constants'
+import { Typography } from '@material-ui/core'
+
+const EVENT_PROGRESS_GOAL = 10
+
+const FlexibleXYPlot = makeWidthFlexible(XYPlot)
+
+const classes = {
+  container: {
+    display: 'flex',
+    margin: '12px 0 20px 0',
+    alignItems: 'center'
+  }
+}
+
+const getData = () => {
+  const data = {}
+  const date = new Date()
+  let month = date.getMonth()
+  let year = date.getFullYear()
+  for (let i = 0; i < 6; i++) {
+    data[`${year} ${month}`] = 0
+    --month
+    if (month < 0) {
+      month = 11
+      --year
+    }
+  }
+  return data
+}
+
+const sixMonthsAgo = () => {
+  const today = new Date()
+  const month = today.getMonth() - 5
+  if (month >= 0) {
+    const year = today.getFullYear()
+    return new Date(year, month)
+  } else {
+    const year = today.getFullYear() - 1
+    return new Date(year, 11 - month)
+  }
+}
+
+const UserProgress = ({ registeredEvents, events }) => {
+  const registeredEventIDs = registeredEvents && registeredEvents.map(event => {
+    return event.eventID
+  })
+
+  // Find events user is registered for
+  const filteredEvents = events && events.filter(event => {
+    return registeredEventIDs.includes(event.id)
+  })
+
+  const data = getData()
+
+  // Fill in data object
+  filteredEvents && filteredEvents.forEach(event => {
+    const parsedDate = new Date(event.startDate)
+    const year = parsedDate.getFullYear()
+    const month = parsedDate.getMonth()
+    data[`${year} ${month}`] += 1
+  })
+
+  // convert 'date strings' to timestamped react-vis data
+  const timestampData = Object.entries(data).map(([key, value]) => {
+    const yearMonthToTimestamp = (yearMonth) => {
+      const year = yearMonth.split(' ')[0]
+      const month = yearMonth.split(' ')[1]
+      return new Date(year, month)
+    }
+
+    return {
+      x: yearMonthToTimestamp(key),
+      y: value
+    }
+  })
+
+  return (
+    <React.Fragment>
+      <div style={classes.container}>
+        <Typography style={{ marginRight: '14px' }}>Events Attended</Typography>
+        <Typography style={{ marginRight: '14px' }}><span style={{ color: COLOR.BIZTECH_GREEN }}>{registeredEventIDs.length}</span>/{EVENT_PROGRESS_GOAL}</Typography>
+        <LinearProgress
+          style={{ flex: 2 }}
+          variant='determinate'
+          value={Math.max(registeredEventIDs.length * EVENT_PROGRESS_GOAL, EVENT_PROGRESS_GOAL)}
```

Comment:
i think the progress bar is based on attended events (checked in), not registered right?

Reply:
how come we need to take the max of registeredEventIDs.length * EVENT_PROGRESS_GOAL and EVENT_PROGRESS_GOAL? what if registeredEventIDs.length * EVENT_PROGRESS_GOAL > 100?